### PR TITLE
Honor the column max length property while populating min/max values for column metadata

### PR DESCRIPTION
**Pull request by raghavyadav01**
Currently addColumnMinMaxValueInfo in SegmentColumnIndexCreator does not honor the column field maxLength for adding min/max values in metadata. If minString for a column is longer than default max limit 512 then minValue is set to null. This causes the segment to be preProcessed every time segment is reloaded. 

Here is the summary of changes:
- write - Use column maxLength property to add min/max in addColumnMinMaxValueInfo.
- load -  check min / max separately and set them separately into the builder and populate accordingly in ColumneMetadataImpl.
- Test -  Added a unit test to reproduce the issue and verify the fix.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Good catch! As discussed offline, changing load time only to treat min/max separately should be good enough. We don't want to use max length in field spec because user might set a huge value in order to ingest data blob, which can make metadata unreadable

**Comment by raghavyadav01** [comment]
Thanks for the feedback @Jackie-Jiang  I have incorporated the review comment and updated the PR.

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java [hunk 1]
```diff
@@ -264,39 +264,9 @@ public static ColumnMetadataImpl fromPropertiesConfiguration(String column, Prop
...
-        case BYTES:
-          builder.setMinValue(BytesUtils.toByteArray(minString));
-          builder.setMaxValue(BytesUtils.toByteArray(maxString));
-          break;
-        default:
-          throw new IllegalStateException("Unsupported data type: " + dataType + " for column: " + column);
-      }
+    // Set min/max value if available
+    if (minString != null || maxString != null) {
+      builder.setColumnMinMaxValues(builder, storedType, column, minString, maxString);
```
Comment (Jackie-Jiang) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java:269]: IMO it is more readable this way:
```suggestion
    // Set min/max value if available
    if (minString != null) {
      builder.setMinValue(parseValue(storedType, column, minString));
    }
    if (maxString != null) {
      builder.setMaxValue(parseValue(storedType, column, maxString));
    }
```

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java (hunk 1 above)
Comment (raghavyadav01) [reply pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java]: Fixed it.

**Pull request merged**